test/pkg/environment/aws/expectations.go

```diff
@@ -563,3 +571,105 @@ func ExpectCapacityReservationsCanceled(ctx context.Context, ec2api *ec2.Client,
 		Expect(err).ToNot(HaveOccurred())
 	}
 }
+
+func (env *Environment) EventuallyExpectRoleCreated(roleName string) {
```

Comment:
I would leave a comment on top of the function along these lines since this does more than just create the role:

> EventuallyExpectRoleCreated creates a role with the provided name. The appropriate policies and trust policy are configured to ensure nodes with this role may join the cluster. Additionally, an AccessEntry is created for the role to ensure nodes are authorized to join the cluster.

I'd add a similar comment for DeleteRole as well.